operator/src/main/java/org/projectnessie/operator/resource/NessieAwsCredentials.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import io.fabric8.generator.annotation.Default;
+import io.fabric8.kubernetes.api.model.LocalObjectReference;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable(builderPackage = "io.fabric8.kubernetes.api.builder", editableEnabled = false)
+@JsonInclude(Include.NON_NULL)
+public record NessieAwsCredentials(
+    @Default("{ \"name\": \"awscreds\" }") LocalObjectReference secretRef,
+    @Default("aws_access_key_id") String awsAccessKeyId,
+    @Default("aws_secret_access_key") String awsSecretAccessKey) {
+
+  public NessieAwsCredentials() {
+    this(null, null, null);
+  }
+
+  public NessieAwsCredentials {
+    if (secretRef == null) {
+      secretRef = new LocalObjectReference("awscreds");
+    }
+    if (awsAccessKeyId == null) {
+      awsAccessKeyId = "aws_access_key_id";
```

Comment:
Wonder whether the defaults in the record constructors and in the `@Default` annotations are redundant. I.e. whether the  `@Default` annotations are superfluous??

Reply:
`@Default` and other annotation from `io.fabric8` modify the generated OpenAPI Schema for the CRD, e.g.:

```yaml
awsAccessKeyId:
  default: aws_access_key_id
  type: string
```

They also influence how the API server sets default values for missing objects in requests received from the client. That's where we need to be careful not to hit NPEs. I will try to remove the constructors and check if no NPEs are being thrown.